### ISSUE TITLE
Publish Developer Snapshots on MavenCentral

### ISSUE THREAD

**Issue by stephanrauh** [TheCoder4eu/BootsFaces-OSP#151]
Every once in a while, I upload an untested developer snapshot of BootsFaces to the Sonatype repository, so you can test bug fixes and new features early.

**Important: Do not use these snapshots in production!**

https://oss.sonatype.org/content/repositories/snapshots/net/bootsfaces/bootsfaces/0.8.2-SNAPSHOT/
https://oss.sonatype.org/content/repositories/snapshots/net/bootsfaces/bootsfaces/0.9.0-SNAPSHOT/

You can access this in your pom.xml after adding a new repository:
``` xml
     <repositories>
        <repository>
          <id>central</id>
          <url>https://oss.sonatype.org/content/repositories/snapshots</url>
          <releases><enabled>true</enabled></releases>
          <snapshots><enabled>true</enabled></snapshots>
        </repository>
    </repositories>
```

The download coordinates of the 0.8.2 snapshot version are:
``` xml
<dependency>
	<groupId>net.bootsfaces</groupId>
	<artifactId>bootsfaces</artifactId>
	<version>0.8.2-SNAPSHOT</version>
	<scope>compile</scope>
</dependency>
```

To get the 0.9.0 snapshot, use these coordinates:
``` xml
<dependency>
	<groupId>net.bootsfaces</groupId>
	<artifactId>bootsfaces</artifactId>
	<version>0.9.0-SNAPSHOT</version>
	<scope>compile</scope>
</dependency>
```

**Comment by stephanrauh** [comment]
I've uploaded a new snapshot covering the BootsFaces Search Expressions (see a live demo of the snapshot at http://www.bootsfaces.net/BootsFacesWeb/forms/searchExpressions.jsf).

**Comment by stephanrauh** [comment]
Uploaded a new version fixing a Search Expression Issue and CSS errors.

**Comment by AlexanderBl4h** [comment]
I am using the current version of BootsFaces to build an application for some students to create articles. There are not more than 100 users active and 20 users logged in at the same time. This app is for the intranet only and not live, yet. I guess, it will get online in the next couple weaks to test it wisely. 

I really want to use the new version, because there are some (for me important) bug fixes. Also, it does feel more `right` to me and  my prof loves to see no bugs (Who does?). 

Can I safely use the newest snapshot until the new version is out? I am using BootsFaces 0.7.0 at the moment.

I mean, if there is some error in the snapshot, I can easily downgrade at the morning, because there are not much users online (just worst case). 

Have you other hints for me? I hope, this is the right place to ask, SO seems the wrong place to me.

**Comment by stephanrauh** [comment]
There's an alternative. You know, I took the snapshot for the sole purpose of finding bugs early. If it works for you, that's great, but there are no guarantees it doesn't contains errors. Use at own risk. That's always true, but in the case of developer snapshots, we didn't even test much. It's a bit risky to run a productive application on such a version without taking precautions.

In the meantime we've been busy developing things like AJAX. That's a major ticket, so it's inevitable we've introduced some errors while fixing others. The next snapshot will contain the AJAX features, so the snapshot you're using will be lost, of sorts. It's still there in the Sonatype repository, but you have to download it manually, because Maven always pulls the latest version.

The alternative is to checkout the "historic" version that works for you from the source code repository and build the library from source. It's just a Maven project, so it's easy. The advantage: If you run into errors, you can fix them yourself, or you can merge a bug fix from GitHub manually without having to accept every change we've done in the meantime. Plus, your source code version doesn't change when I upload the next snapshot :).

That said, I suppose we'll finish the AJAX and search expression tickets soon and start to concentrate on bug fixing. Chances are one of the next snapshots is as stable as the snapshot you're currently using.

**Comment by stephanrauh** [comment]
@AlexanderBl4h I'd like to upload a new snapshot soon. It includes the new AJAX functionality, so chances are it's less stable than its predecessor. Is that a problem to your project?

**Comment by AlexanderBl4h** [comment]
No. I am currently using still 0.7.0. If I have time, I will test the new snapshot a bit, with my project.

**Comment by stephanrauh** [comment]
I just uploaded a new developer preview snapshot. I've started to implement the new unified AJAX, which is one of the big tickets, so I'm sure the version is buggy. Which makes it even more important to put it under test. If you find an error, please report it to use. Thanks!

As for the new AJAX engine: my AJAX example show what's already possible (and everything that's not shown there has probably not been implemented yet :)): https://github.com/stephanrauh/BootsFaces-Examples/tree/master/AJAX

**Comment by stephanrauh** [comment]
I've just uploaded a new developer snapshot. It contains the finished unified AJAX and a few bug fixes. It took some time to implement the AJAX engine, so it's likely I've introduced bug along the way. If you find an error, please report it to us. Thanks!

**Comment by stephanrauh** [comment]
I've updated the developer snapshot. It contains a couple of minor bugfixes and add CSS classes to labels and input fields, depending on whether it's a mandatory field, or whether it has an error message (see #190 ).

**Comment by stephanrauh** [comment]
Implemented #197 and uploaded a new snapshot version.

**Comment by stephanrauh** [comment]
Uploaded a new version containing a couple of bug fixes.

**Comment by stephanrauh** [comment]
Uploaded a new snapshot cotaining a bugfix concerning b:textArea (#201).

**Comment by stephanrauh** [comment]
Uploaded a new snapshot containing a bugfix concerning standard JSF AJAX.

**Comment by stephanrauh** [comment]
Just uploaded the snapshot of the 0.8.1 version.

**Comment by stephanrauh** [comment]
Just in case someone want to help us testing: I've published a new developer snapshot of BootsFaces 0.8.1 today, and we plan to publish this version really soon now.

**Comment by stephanrauh** [comment]
Uploaded a developer snapshot of version 0.8.2, fixing #263 and the bug calling actions twice when a command button has been clicked. Note that the command button bit is still work in progress. Use this snapshot on own risk!

**Comment by stephanrauh** [comment]
Uploaded a new developer snapshot.

**Comment by stephanrauh** [comment]
Just uploaded another developer snapshot of 0.8.2. We'd like to publish this version soon, so if you've got some spare time to test the new snapshot, we'd appreciate this very much. Thanks in advance!

**Comment by stephanrauh** [comment]
Just uploaded another developer snapshot of 0.8.2.

**Comment by TheCoder4eu** [comment]
@stephanrauh you have a go! ;-)

**Comment by stephanrauh** [comment]
Today, I've released the final release 0.8.5 of BootsFaces on Maven Central.
This ticket already has a very long ticket. Therefore, I've decided to replace it by a follow-up ticket. Dear subscribers, please follow the new ticket #369. Thanks!
